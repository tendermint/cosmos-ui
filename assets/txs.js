module.exports = [
  {
    hash: "not a real hash",
    time: 1042 + 1000, // set by Voyager
    height: "3436",
    tx: {
      type: "8EFE47F0625DE8",
      value: {
        msg: [
          {
            type: "cosmos-sdk/Send",
            value: {
              inputs: [
                {
                  coins: [
                    {
                      denom: "jbcoins",
                      amount: 1234
                    }
                  ],
                  address: "tb1da6xsetjg9jxgun9wdesexv05j"
                }
              ],
              outputs: [
                {
                  coins: [
                    {
                      denom: "jbcoins",
                      amount: 1234
                    }
                  ],
                  address: "tb1d4u5zerywfjhxuc9nudvw"
                }
              ]
            }
          }
        ],
        fee: {
          amount: [
            {
              denom: "",
              amount: 0
            }
          ],
          gas: 500000
        },
        gas: 500000,
        signatures: [
          {
            pub_key: {
              type: "AC26791624DE60",
              value: "70df5mUA3bFTJheAI6umXvABNuUU+q/3nOPS6lmT3tU="
            },
            signature: {
              type: "6BF5903DA1DB28",
              value:
                "oODzZz2t1u678Jhy2zFnFbBu7S2IP1zjOIoyVDOvGkQ6WkiiRF3zxEJbNAjHGdDjXf7/NQpMOtpt8FC1x6e6DA=="
            },
            account_number: 28,
            sequence: 0
          }
        ]
      },
      result: {
        gasUsed: "3324",
        tags: [
          {
            key: "c2VuZGVy",
            value: "N0ZFQzVEREE0NDA3RDVBRUQ5OTlFMTM3QjYxQTQ2QTEzQTc2MkJCNg=="
          },
          {
            key: "cmVjaXBpZW50",
            value: "N0ZFQzVEREE0NDA3RDVBRUQ5OTlFMTM3QjYxQTQ2QTEzQTc2MkJCNg=="
          }
        ],
        fee: {}
      }
    }
  },
  {
    hash: "not a real hash2",
    time: 1042 + 1010, // set by Voyager
    height: "3438",
    tx: {
      type: "8EFE47F0625DE8",

      value: {
        msg: [
          {
            type: "cosmos-sdk/Send",
            value: {
              inputs: [
                {
                  coins: [
                    {
                      denom: "jbcoins",
                      amount: 1234
                    }
                  ],
                  address: "tb1d4u5zerywfjhxuc9nudvw"
                }
              ],
              outputs: [
                {
                  coins: [
                    {
                      denom: "jbcoins",
                      amount: 1234
                    }
                  ],
                  address: "tb1da6xsetjg9jxgun9wdesexv05j"
                }
              ]
            }
          }
        ],
        fee: {
          amount: [
            {
              denom: "",
              amount: 0
            }
          ],
          gas: 500000
        },
        gas: 500000,
        signatures: [
          {
            pub_key: {
              type: "AC26791624DE60",
              value: "70df5mUA3bFTJheAI6umXvABNuUU+q/3nOPS6lmT3tU="
            },
            signature: {
              type: "6BF5903DA1DB28",
              value:
                "oODzZz2t1u678Jhy2zFnFbBu7S2IP1zjOIoyVDOvGkQ6WkiiRF3zxEJbNAjHGdDjXf7/NQpMOtpt8FC1x6e6DA=="
            },
            account_number: 28,
            sequence: 0
          }
        ]
      },
      result: {
        gasUsed: "3324",
        tags: [
          {
            key: "c2VuZGVy",
            value: "N0ZFQzVEREE0NDA3RDVBRUQ5OTlFMTM3QjYxQTQ2QTEzQTc2MkJCNg=="
          },
          {
            key: "cmVjaXBpZW50",
            value: "N0ZFQzVEREE0NDA3RDVBRUQ5OTlFMTM3QjYxQTQ2QTEzQTc2MkJCNg=="
          }
        ],
        fee: {}
      }
    }
  },
  {
    hash: "not a real hash3",

    time: 1042 + 100, // set by Voyager
    height: "466",
    tx: {
      type: "8EFE47F0625DE8",
      value: {
        msg: [
          {
            type: "cosmos-sdk/Send",
            value: {
              inputs: [
                {
                  coins: [
                    {
                      denom: "jbcoins",
                      amount: 1234
                    },
                    {
                      denom: "fabocoins",
                      amount: 1
                    },
                    {
                      denom: "mattcoins",
                      amount: 42
                    }
                  ],
                  address: "tb1da6xsetjg9jxgun9wdesexv05j"
                }
              ],
              outputs: [
                {
                  coins: [
                    {
                      denom: "jbcoins",
                      amount: 1234
                    },
                    {
                      denom: "fabocoins",
                      amount: 1
                    },
                    {
                      denom: "mattcoins",
                      amount: 42
                    }
                  ],
                  address: "tb1da6xsetjg9jxgun9wdesexv05j"
                }
              ]
            }
          }
        ],
        fee: {
          amount: [
            {
              denom: "",
              amount: 0
            }
          ],
          gas: 500000
        },
        gas: 500000,
        signatures: [
          {
            pub_key: {
              type: "AC26791624DE60",
              value: "70df5mUA3bFTJheAI6umXvABNuUU+q/3nOPS6lmT3tU="
            },
            signature: {
              type: "6BF5903DA1DB28",
              value:
                "oODzZz2t1u678Jhy2zFnFbBu7S2IP1zjOIoyVDOvGkQ6WkiiRF3zxEJbNAjHGdDjXf7/NQpMOtpt8FC1x6e6DA=="
            },
            account_number: 28,
            sequence: 0
          }
        ]
      },
      result: {
        gasUsed: "3324",
        tags: [
          {
            key: "c2VuZGVy",
            value: "N0ZFQzVEREE0NDA3RDVBRUQ5OTlFMTM3QjYxQTQ2QTEzQTc2MkJCNg=="
          },
          {
            key: "cmVjaXBpZW50",
            value: "N0ZFQzVEREE0NDA3RDVBRUQ5OTlFMTM3QjYxQTQ2QTEzQTc2MkJCNg=="
          }
        ],
        fee: {}
      }
    }
  },
  {
    hash: "not a real hash4",

    time: 1042 + 150, // set by Voyager
    height: "566",
    tx: {
      type: "8EFE47F0625DE8",
      value: {
        msg: [
          {
            type: "cosmos-sdk/MsgDelegate",
            value: {
              validator_addr:
                "cosmosvaladdr15ky9du8a2wlstz6fpx3p4mqpjyrm5ctqzh8yqw",
              delegation: {
                amount: 4200,
                denom: "steak"
              }
            }
          }
        ],
        fee: {
          amount: [
            {
              denom: "",
              amount: 0
            }
          ],
          gas: 500000
        },
        gas: 500000,
        signatures: [
          {
            pub_key: {
              type: "AC26791624DE60",
              value: "70df5mUA3bFTJheAI6umXvABNuUU+q/3nOPS6lmT3tU="
            },
            signature: {
              type: "6BF5903DA1DB28",
              value:
                "oODzZz2t1u678Jhy2zFnFbBu7S2IP1zjOIoyVDOvGkQ6WkiiRF3zxEJbNAjHGdDjXf7/NQpMOtpt8FC1x6e6DA=="
            },
            account_number: 28,
            sequence: 0
          }
        ]
      },
      result: {
        gasUsed: "3324",
        tags: [
          {
            key: "c2VuZGVy",
            value: "N0ZFQzVEREE0NDA3RDVBRUQ5OTlFMTM3QjYxQTQ2QTEzQTc2MkJCNg=="
          },
          {
            key: "cmVjaXBpZW50",
            value: "N0ZFQzVEREE0NDA3RDVBRUQ5OTlFMTM3QjYxQTQ2QTEzQTc2MkJCNg=="
          }
        ],
        fee: {}
      }
    }
  },
  {
    hash: "not a real hash5",

    time: 1042 + 150, // set by Voyager
    height: "566",
    tx: {
      type: "8EFE47F0625DE8",
      value: {
        msg: [
          {
            type: "cosmos-sdk/BeginUnbonding",
            value: {
              validator_addr:
                "cosmosvaladdr15ky9du8a2wlstz6fpx3p4mqpjyrm5ctqzh8yqw",
              shares: 100
            }
          }
        ],
        fee: {
          amount: [
            {
              denom: "",
              amount: 0
            }
          ],
          gas: 500000
        },
        gas: 500000,
        signatures: [
          {
            pub_key: {
              type: "AC26791624DE60",
              value: "70df5mUA3bFTJheAI6umXvABNuUU+q/3nOPS6lmT3tU="
            },
            signature: {
              type: "6BF5903DA1DB28",
              value:
                "oODzZz2t1u678Jhy2zFnFbBu7S2IP1zjOIoyVDOvGkQ6WkiiRF3zxEJbNAjHGdDjXf7/NQpMOtpt8FC1x6e6DA=="
            },
            account_number: 28,
            sequence: 0
          }
        ]
      },
      result: {
        gasUsed: "3324",
        tags: [
          {
            key: "c2VuZGVy",
            value: "N0ZFQzVEREE0NDA3RDVBRUQ5OTlFMTM3QjYxQTQ2QTEzQTc2MkJCNg=="
          },
          {
            key: "cmVjaXBpZW50",
            value: "N0ZFQzVEREE0NDA3RDVBRUQ5OTlFMTM3QjYxQTQ2QTEzQTc2MkJCNg=="
          }
        ],
        fee: {}
      }
    }
  }
]
