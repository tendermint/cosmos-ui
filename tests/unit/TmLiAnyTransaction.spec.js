import { mount } from "@vue/test-utils"
import { TmLiAnyTransaction } from "@/index"
import txs from "../../assets/txs.js"

describe("TmLiAnyTransaction", () => {
  let wrapper
  let propsData = {
    transaction: txs[0],
    validators: [
      {
        owner: "cosmosvaladdr15ky9du8a2wlstz6fpx3p4mqpjyrm5ctqzh8yqw",
        description: {
          moniker: "cool validator"
        }
      }
    ],
    address: "tb1da6xsetjg9jxgun9wdesexv05j"
  }

  beforeEach(() => {
    wrapper = mount(TmLiAnyTransaction, {
      propsData
    })
  })

  it("shows bank transactions", () => {
    expect(wrapper.vm.$el).toMatchSnapshot()
  })

  it("shows stake transactions", () => {
    wrapper.setProps({
      transaction: txs[3]
    })
    expect(wrapper.vm.$el).toMatchSnapshot()
  })

  it("shows unknown transactions", () => {
    let unknownTx = JSON.parse(JSON.stringify(txs[0]))
    unknownTx.tx.value.msg[0].type = "UNKNOWN"
    wrapper.setProps({
      transaction: unknownTx
    })
    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})
