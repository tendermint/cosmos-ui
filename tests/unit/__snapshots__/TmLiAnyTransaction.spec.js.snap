// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TmLiAnyTransaction shows bank transactions 1`] = `
<div
  class="tm-li-tx"
>
  <div
    class="tm-li-tx__icon"
  >
    <img
      src="../../assets/cosmos-logo.png"
      style="border-color: #25293f;"
    />
  </div>
  <div
    class="tm-li-tx__content"
  >
    <p
      class="tm-li-tx__content__caption"
    >
      <div>
        Sent 
        <b>
          1234
        </b>
        <span>
           JBCOINS
        </span>
      </div>
    </p>
    <div
      class="tm-li-tx__content__information"
    >
      <div
        class="tm-li-tx__content__information__details"
      >
        <span>
          To tb1d4u5zerywfjhxuc9nudvw
        </span>
      </div>
      <div
        class="tm-li-tx__content__information__block"
      >
        <a
          href="https://explorecosmos.network/blocks/3436"
        >
          Block #3436 
        </a>
        @ 2:03 am
      </div>
    </div>
  </div>
</div>
`;

exports[`TmLiAnyTransaction shows stake transactions 1`] = `
<div
  class="tm-li-tx"
>
  <div
    class="tm-li-tx__icon"
  >
    <img
      src="../../assets/cosmos-logo.png"
      style="border-color: #512584;"
    />
  </div>
  <div
    class="tm-li-tx__content"
  >
    <p
      class="tm-li-tx__content__caption"
    >
      <div>
        Delegated 
        <b>
          4200
        </b>
        <span>
           steak
        </span>
      </div>
    </p>
    <div
      class="tm-li-tx__content__information"
    >
      <div
        class="tm-li-tx__content__information__details"
      >
        <div>
          To 
          <a
            href="/cosmosvaladdr15ky9du8a2wlstz6fpx3p4mqpjyrm5ctqzh8yqw"
          >
            cool validator
          </a>
        </div>
      </div>
      <div
        class="tm-li-tx__content__information__block"
      >
        <a
          href="https://explorecosmos.network/blocks/566"
        >
          Block #566 
        </a>
        @ 2:03 am
      </div>
    </div>
  </div>
</div>
`;

exports[`TmLiAnyTransaction shows unknown transactions 1`] = `
<div
  class="tm-li-tx"
>
  <div
    class="tm-li-tx__icon"
  >
    <img
      src="../../assets/cosmos-logo.png"
    />
  </div>
  <div
    class="tm-li-tx__content"
  >
    <p
      class="tm-li-tx__content__caption"
    >
      <span>
        Unknown Transaction Type
      </span>
    </p>
    <div
      class="tm-li-tx__content__information"
    >
      <div
        class="tm-li-tx__content__information__details"
      />
      <div
        class="tm-li-tx__content__information__block"
      >
        <a
          href="https://explorecosmos.network/blocks/3436"
        >
          Block #3436 
        </a>
        @ 2:03 am
      </div>
    </div>
  </div>
</div>
`;
